java/src/jmri/jmrix/loconet/messageinterp/LocoNetMessageInterpret.java

```diff
@@ -2323,6 +2349,157 @@ private static String interpretOpcMultiSenseTranspPresence(LocoNetMessage l, Str
+         *  bit 0 RailCom Dynamic Value high bit
+         *
+         * <RCDV_L> {6:0} represent the lower 7 bits of the 8 bit RailCom Dynamic Value.  The 8th bit is bit 0 of <RCDV_H>
+         *
+         * <CHK>
+         *
+         * Information reverse-engineered by Michael Ricahrdson
+         */
+
+        if (l.getElement(1) == 0x09){  // Only process 0xE0 0x09 messages
+            // Transponding Event
+            // get system and user names
+
+            int type = l.getElement(2) & LnConstants.OPC_MULTI_SENSE_MSG;
+            //0x00 = absent
+            //0x20 = present
+            //0x40 = present with App Dyn
+            //0x60 = unknown
+
+            if (type == 0x60) { //uknown at this point
+                return Bundle.getMessage("LN_MSG_OPC_MULTI_SENSE_LONG_UNKNOWN_MESSAGE");
+
+            } else {  //Process 0x00, 0x20, and 0x40
+                String reporterSystemName = reporterPrefix + ((l.getElement(2) & 0x1F) * 128 + l.getElement(3) + 1);
+
+                Reporter reporter = InstanceManager.getDefault(ReporterManager.class).getReporter(reporterSystemName);
+                String reporterUserName = "";
+                if (reporter != null) {
+                    String uname = reporter.getUserName();
+                    if ((uname != null) && (!uname.isEmpty())) {
+                        reporterUserName = uname;
+                    }
+                }
+
+                String locoAddr = convertToMixed(l.getElement(5), l.getElement(4));
+
+                String transpActivity = (type == LnConstants.OPC_MULTI_SENSE_ABSENT)
+                        ? Bundle.getMessage("LN_MSG_OPC_MULTI_SENSE_TRANSP_HELPER_IS_ABSENT")
+                        : Bundle.getMessage("LN_MSG_OPC_MULTI_SENSE_TRANSP_HELPER_IS_PRESENT");
+
+                String direction = ((l.getElement(6) & 0x40) == 0)
+                        ? Bundle.getMessage("LN_MSG_OPC_MULTI_SENSE_LONG_LOCO_DIRECTION_HELPER_EAST")
+                        : Bundle.getMessage("LN_MSG_OPC_MULTI_SENSE_LONG_LOCO_DIRECTION_HELPER_WEST");
+
+                if (type == LnConstants.OPC_MULTI_SENSE_RAILCOM_AD) {
+                    int indexValue = (l.getElement(6) & 0x3E)/2; //bits 5-1
+                    int dynamicValue = l.getElement(7) + (l.getElement(6) & 0x01) * 128;
+
+                    String railcomAdString = convertRailComAD(indexValue, dynamicValue);
+                    String multiSenseLongString = Bundle.getMessage("LN_MSG_OPC_MULTI_SENSE_LONG_TRANSP_REPORT",
+                            locoAddr, direction, transpActivity, reporterSystemName, reporterUserName);
+
+                    return Bundle.getMessage("LN_MSG_OPC_MULTI_SENSE_LONG_TRANSP_RAILCOM_REPORT",
+                            multiSenseLongString, railcomAdString);
+
+                } else {
+                    return Bundle.getMessage("LN_MSG_OPC_MULTI_SENSE_LONG_TRANSP_REPORT",
```

Comment:
It seems that this return path does not attempt to look at <RCVD_H> bits 5-0.  Is this intentional?

Reply:
Intentional.  Not only is this specific return path not looking at <RCVD_H> bits 5-0, but also not looking at all bits of <RCDV_L>.  

From all the packets I have captured and examined, when <MSL_I> (byte 2) bits 6-5 are either 00/0x00 (Absent) or 01/0x20 (Present), the value of <RCVD_H> bits 5-0 and all of <RCDV_L> are always 0.  To my current understanding, these 13 bits in total, seem to be only used when When <MSL_I> (byte 2) bits 6-5 is 10/0x60 (Present with AppDyn Message) 

Is the approach to "ignore" OK, or would you recommend checking, and if non-zero sending an unknown report?  probably - `Bundle.getMessage("LN_MSG_OPC_MULTI_SENSE_LONG_UNKNOWN_MESSAGE");`

Reply:
Now handled with the latest code and your May 4 feedback above . For Absent (0x00) or present (0x20) type reports `if ((l.getElement(6) & 0x3F) != 0 || l.getElement(7) != 0)` I now use LN_MSG_OPC_MULTI_SENSE_LONG_UNKNOWN_MESSAGE.  

I also added test cases for this unexpected case and additional test cases with minimum and maximum values.